src/iosource/pcap/Source.cc

```diff
@@ -84,16 +84,57 @@ void PcapSource::OpenLive()
 		props.netmask = PktSrc::NETMASK_UNKNOWN;
 #endif
 
-	// We use the smallest time-out possible to return almost immediately if
+	// ### We use the smallest time-out possible to return almost immediately if
 	// no packets are available. (We can't use set_nonblocking() as it's
 	// broken on FreeBSD: even when select() indicates that we can read
 	// something, we may get nothing if the store buffer hasn't filled up
 	// yet.)
-	pd = pcap_open_live(props.path.c_str(), SnapLen(), 1, 1, tmp_errbuf);
```

Comment:
-	pd = pcap_open_live(props.path.c_str(), SnapLen(), 1, 1, tmp_errbuf);
rsmmr added a note
7 days ago
Can you elaborate why you replaced the pcap_open_live call with the individual calls below? Does this raise the version of libpcap required?
knielander added a note
4 days ago
It doesn't raise the version, but it will allow you to specify the ring buffer size.